src/main/java/com/github/dockerjava/core/DefaultDockerClientConfig.java

```diff
@@ -0,0 +1,467 @@
+    }
+
+    @Override
+    public String getRegistryUrl() {
+        return registryUrl;
+    }
+
+    public String getDockerConfig() {
+        return dockerConfig;
+    }
+
+    private AuthConfig getAuthConfig() {
+        AuthConfig authConfig = null;
+        if (getRegistryUsername() != null && getRegistryPassword() != null && getRegistryEmail() != null
+                && getRegistryUrl() != null) {
+            authConfig = new AuthConfig()
+                    .withUsername(getRegistryUsername())
+                    .withPassword(getRegistryPassword())
+                    .withEmail(getRegistryEmail())
+                    .withRegistryAddress(getRegistryUrl());
+        }
+        return authConfig;
+    }
+
+    @Override
+    public AuthConfig effectiveAuthConfig(String imageName) {
+        AuthConfig authConfig = null;
+
+        File dockerCfgFile = new File(getDockerConfig() + File.separator + DOCKER_CFG);
+
+        if (dockerCfgFile.exists() && dockerCfgFile.isFile() && imageName != null) {
+            AuthConfigFile authConfigFile;
+            try {
+                authConfigFile = AuthConfigFile.loadConfig(dockerCfgFile);
+            } catch (IOException e) {
+                throw new DockerClientException("Failed to parse dockerCfgFile", e);
+            }
+            ReposTag reposTag = NameParser.parseRepositoryTag(imageName);
+            HostnameReposName hostnameReposName = NameParser.resolveRepositoryName(reposTag.repos);
+
+            authConfig = authConfigFile.resolveAuthConfig(hostnameReposName.hostname);
+        }
+
+        AuthConfig otherAuthConfig = getAuthConfig();
+
+        if (otherAuthConfig != null) {
+            authConfig = otherAuthConfig;
+        }
+
+        return authConfig;
+    }
+
+    @Override
+    public AuthConfigurations getAuthConfigurations() {
+        File dockerCfgFile = new File(getDockerConfig() + File.separator + DOCKER_CFG);
+        if (dockerCfgFile.exists() && dockerCfgFile.isFile()) {
+            AuthConfigFile authConfigFile;
+            try {
+                authConfigFile = AuthConfigFile.loadConfig(dockerCfgFile);
+            } catch (IOException e) {
+                throw new DockerClientException("Failed to parse dockerCfgFile", e);
+            }
+
+            return authConfigFile.getAuthConfigurations();
+        }
+
+        return new AuthConfigurations();
+    }
+
+    @Override
+    public SSLContext getSSLContext() {
+        return sslContext;
+    }
+
+    // CHECKSTYLE:OFF
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        DefaultDockerClientConfig that = (DefaultDockerClientConfig) o;
+
+        return EqualsBuilder.reflectionEquals(this, that);
+    }
+
+    @Override
+    public int hashCode() {
+        return HashCodeBuilder.reflectionHashCode(this);
+    }
+
+    // CHECKSTYLE:ON
```

Comment:
checkstyle was offed only because of ifs in equals that was generated with IDE. With *Builders checkstyle offs could be removed.